### ISSUE TITLE
+= does not work on multi-dim arrays in some situtations.

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#3226]
`+=` does not work on multi-dim arrays in some situations.

### compare

https://dev.deck.net/99b86b4650a88350374349224b817a92

### to

https://dotnetfiddle.net/kHmyZb

```csharp
using System;
using System.Collections.Generic;

public static class Program
{
    public static void Main()
    {
        var a = new int[3, 3];

        a[0, 0] = 1;

        for (var y = 0; y < (a.GetLength(1) - 1); y++)
            for (var x = 0; x < (a.GetLength(0) - 1); x++)
                a[x + 1, y + 1] += a[x, y];

        List<int> list = new List<int>();
        string s = "";

        foreach (var i in a)
            s += i;

        Console.WriteLine("Expected: " + 100010001);
        Console.WriteLine("Actual: " + s);
        Console.WriteLine("100010001"  == s ? "Passed" : "Failed");
    }
}
```

**Comment by geoffreymcgill** [comment]
Hi @michaelcheers – Thanks for this defect report and excellent sample demonstrating how to reproduce. We will investigate and attempt to fix right away.

**Comment by fabriciomurta** [comment]
Merged upstream. The fix made it to Bridge 16.5.0!
